app/connectors/MarriageAllowanceDESConnector.scala

```diff
@@ -0,0 +1,179 @@
+import play.api.http.Status._
+import play.api.libs.json.{JsPath, JsValue}
+import play.api.{Configuration, Logger, Play}
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.play.config.ServicesConfig
+import utils.WSHttp
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+
+trait MarriageAllowanceDESConnector extends MarriageAllowanceConnector with DESResponseCodes {
+
+  private def handleValidationError[A]: Seq[(JsPath, scala.Seq[ValidationError])] => Left[DataRetrievalError, A] =  err => {
+
+    val extractValidationErrors: Seq[(JsPath, scala.Seq[ValidationError])] => String = errors => {
+      errors.map {
+        case (path, List(validationError: ValidationError, _*)) => s"$path: ${validationError.message}"
+      }.mkString(", ").trim
+    }
+
+    Logger.error(s"Not able to parse the response received from DES with error ${extractValidationErrors(err)}")
+    Left(ResponseValidationError)
+  }
+
+
+  def findCitizen(nino: Nino)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[JsValue] = {
+    val path = url(s"/marriage-allowance/citizen/${nino}")
+    httpGet.GET[JsValue](path)(implicitly, buildHeaderCarrier(hc), ec)
+  }
+
+  def listRelationship(cid: Cid, includeHistoric: Boolean = true)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[JsValue] = {
+    val path = url(s"/marriage-allowance/citizen/${cid}/relationships?includeHistoric=${includeHistoric}")
+    httpGet.GET[JsValue](path)
+  }
+
+  def findRecipient(findRecipientRequest: FindRecipientRequest)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Either[DataRetrievalError, UserRecord]] = {
+
+    def evaluateCodes(findRecipientResponseDES: FindRecipientResponseDES): Either[DataRetrievalError, UserRecord] = {
+
+      (findRecipientResponseDES.returnCode, findRecipientResponseDES.reasonCode) match {
+        case(ProcessingOK, ProcessingOK) =>{
+          metrics.incrementSuccessCounter(ApiType.FindRecipient)
+          Right(UserRecord(findRecipientResponseDES.instanceIdentifier, findRecipientResponseDES.updateTimeStamp))
+        }
+        case codes @ (ErrorReturnCode, NinoNotFound) => {
+          val codedErrorResponse = FindRecipientCodedErrorResponse(codes._1, codes._2, "Nino not found and Nino not found in merge trail")
+          Logger.warn(codedErrorResponse.errorMessage)
+          metrics.incrementSuccessCounter(ApiType.FindRecipient)
+          Left(codedErrorResponse)
+        }
+        case codes @ (ErrorReturnCode, MultipleNinosInMergeTrail) => {
+          val codedErrorResponse = FindRecipientCodedErrorResponse(codes._1, codes._2, "Nino not found and Nino found in multiple merge trails")
+          Logger.warn(codedErrorResponse.errorMessage)
+          metrics.incrementSuccessCounter(ApiType.FindRecipient)
+          Left(codedErrorResponse)
+        }
+        case codes @ (ErrorReturnCode, ConfidenceCheck) => {
```

Comment:
I cant find unit tests for this branch of code under the Coded error responses tests